### PR TITLE
Improvement/routeref new

### PR DESCRIPTION
**Pull request by CrowleyRajapakse**
## Purpose
Improvement/routeref new

### CODE REVIEW

File: helm-charts/templates/data-plane/config-deployer/config-ds-deployment.yaml [hunk 1]
```diff
@@ -77,6 +77,9 @@ spec:
             {{- else }}
               subPath: tls.key
             {{- end }}
+            - name: prometheus-jmx-config-volume
```
Comment (sgayangi) [inline helm-charts/templates/data-plane/config-deployer/config-ds-deployment.yaml:80]: Should be removed, already included in a previous line

File: helm-charts/templates/data-plane/gateway-components/common-controller/common-controller-service.yaml [hunk 2]
```diff
@@ -38,4 +38,9 @@ spec:
     - name: https-internal-api
       protocol: TCP
       port: 18003
+    {{ if and .Values.wso2.apk.metrics .Values.wso2.apk.metrics.enabled }}
```
Comment (sgayangi) [inline helm-charts/templates/data-plane/gateway-components/common-controller/common-controller-service.yaml:41]: Can remove the exposed port

File: helm-charts/templates/data-plane/config-deployer/config-ds-service.yaml [hunk 3]
```diff
@@ -28,4 +28,12 @@ spec:
...
       protocol: TCP
       port: 9443
+    {{ if and .Values.wso2.apk.metrics .Values.wso2.apk.metrics.enabled }}
+    - name: prometheus-config-ds-port-lang
+      protocol: TCP
+      port: 18008
+    - name: prometheus-config-ds-port-system
+      protocol: TCP
+      port: 18010
+    {{ end }}
```
Comment (sgayangi) [inline helm-charts/templates/data-plane/config-deployer/config-ds-service.yaml:38]: Should remove this section

File: helm-charts/templates/data-plane/config-deployer/config-ds-deployment.yaml [hunk 4]
```diff
@@ -77,6 +77,9 @@ spec:
...
+              mountPath: /tmp/metrics/prometheus-jmx-config-configds.yml
+              subPath: prometheus-jmx-config-configds.yml
```
Comment (sgayangi) [inline helm-charts/templates/data-plane/config-deployer/config-ds-deployment.yaml:82]: Should remove this section

File: helm-charts/templates/data-plane/gateway-components/common-controller/common-controller-service.yaml [hunk 5]
```diff
@@ -38,4 +38,9 @@ spec:
...
+    - name: prometheus-common-controller-port
+      protocol: TCP
+      port: 18009
+    {{ end }}
```
Comment (sgayangi) [inline helm-charts/templates/data-plane/gateway-components/common-controller/common-controller-service.yaml:45]: Should remove this section

File: common-controller/commoncontroller/common_controller.go [hunk 6]
```diff
@@ -214,6 +214,12 @@ func InitCommonControllerServer(conf *config.Config) {
 
 	loggers.LoggerAPKOperator.Info("Starting common controller ....")
 
+	// Start the metrics server
+	if conf.CommonController.Metrics.Enabled && strings.EqualFold(conf.CommonController.Metrics.Type, metrics.PrometheusMetricType) {
+		loggers.LoggerAPKOperator.Info("Starting Prometheus Metrics Server ....")
+		go metrics.StartPrometheusMetricsServer(conf.CommonController.Metrics.Port)
+	}
+
```
Comment (sgayangi) [inline common-controller/commoncontroller/common_controller.go:222]: Should remove this section: duplicate

File: helm-charts/templates/data-plane/config-deployer/config-ds-service.yaml (hunk 3 above)
Comment (CrowleyRajapakse) [reply helm-charts/templates/data-plane/config-deployer/config-ds-service.yaml]: fixed

File: helm-charts/templates/data-plane/gateway-components/common-controller/common-controller-service.yaml (hunk 5 above)
Comment (CrowleyRajapakse) [reply helm-charts/templates/data-plane/gateway-components/common-controller/common-controller-service.yaml]: fixed

File: helm-charts/templates/data-plane/config-deployer/config-ds-deployment.yaml (hunk 4 above)
Comment (CrowleyRajapakse) [reply helm-charts/templates/data-plane/config-deployer/config-ds-deployment.yaml]: fixed

File: common-controller/commoncontroller/common_controller.go (hunk 6 above)
Comment (CrowleyRajapakse) [reply common-controller/commoncontroller/common_controller.go]: fixed

**Review by Krishanx92 — APPROVED** [review]

**Pull request merged**